src/libraries/System.Threading.Tasks/tests/Task/TaskRtTests.cs

```diff
@@ -691,6 +691,24 @@ public static void RunDelayTests_NegativeCases()
             }
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public static void TaskDelay_Cancellation_ContinuationsInvokedAsynchronously()
+        {
+            var cts = new CancellationTokenSource();
+
+            var tl = new ThreadLocal<int>();
+            Task c = Task.Delay(-1, cts.Token).ContinueWith(_ =>
+            {
+                Assert.Equal(0, tl.Value);
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+
+            tl.Value = 42;
+            cts.Cancel();
+            tl.Value = 0;
```

Comment:
Is the clearing of this `AsyncLocal` significant for this test? I would have thought it's just a no-op.

Reply:
Technically it's a nop.  But I like doing it to make the scoping of the lifetime of the value very explicit.